RoboticonColony/Assets/Editor/Unit_Tests/InventoryTest.cs

```
@@ -0,0 +1,136 @@
+
+        //Check if all resources have initial value set to zero
+        Assert.AreEqual(0, inv.Money);
+        Assert.AreEqual(0, inv.GetItemAmount(ItemType.Ore));
+        Assert.AreEqual(0, inv.GetItemAmount(ItemType.Power));
+        Assert.AreEqual(0, inv.GetItemAmount(ItemType.Roboticon));
+    }
+
+    [Test]
+    public void CreatePopulatedInventory()
+    {
+        //Create a populated inventory instance
+        Inventory inv = new Inventory(4, 3, 2, 1);
+
+        //Check if all resources have the value they were set
+        Assert.AreEqual(4, inv.Money);
+        Assert.AreEqual(3, inv.GetItemAmount(ItemType.Ore));
+        Assert.AreEqual(2, inv.GetItemAmount(ItemType.Power));
+        Assert.AreEqual(1, inv.GetItemAmount(ItemType.Roboticon));
+    }
+
+    [Test]
+    public void AddMoneyToInventory()
+    {
+        //Create an empty inventory instance
+        Inventory inv = new Inventory();
+
+        //Prove that the inventory has no money
+        Assert.AreEqual(0, inv.Money);
+
+        //Now add money
+        inv.AddMoney(6);
+
+        //Check if the money has successfully been added
+        Assert.AreEqual(6, inv.Money);
+    }
+
+    [Test]
+    public void AddItemToInventory()
+    {
+        //Create an empty inventory instance
+        Inventory inv = new Inventory();
+
+        //Prove that the inventory has no ore
+        Assert.AreEqual(0, inv.GetItemAmount(ItemType.Ore));
+
+        //Now add money
+        inv.AddItem(ItemType.Ore, 6);
+
+        //Check if the ore has successfully been added
+        Assert.AreEqual(6, inv.GetItemAmount(ItemType.Ore));
+    }
+
+    [Test]
+    public void FailedMoneyTransaction()
+    {
+        //Create two inventories
+        Inventory inv1 = new Inventory(1, 2, 3, 4);
+        Inventory inv2 = new Inventory(10, 11, 12, 13);
+
+        //Attempt to transfer 2 money from inv1 to inv2, even though inv1 only has 1 money
+        bool successful = inv1.TransferMoney(2, inv2);
```

shouldnt this throw an error rather than return false?